release/v1alpha1/types.go

```diff
@@ -0,0 +1,76 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PromotionTestResults encapsulates the inputs to produce a new set of promotion tests and
+// the status of the promotion tests
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type PromotionTestResults struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Spec the inputs used to create the promotion test
+	Spec PromotionTestSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+
+	// Status is the current status of the promotion test
+	Status PromotionTestResultsStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// PromotionTestSpec has the information to represent a PromotionTest
+type PromotionTestSpec struct {
+	Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
+}
+
+// PromotionTestResultsStatus the status of all the promotion test jobs
+type PromotionTestResultsStatus struct {
+	JobResults []JobResults `json:"results,omitempty" protobuf:"bytes,1,rep,name=results"`
+}
+
+// JobAggregateResultState the final state of the job
+type JobAggregateResultState string
+
+const (
+	// FailedJobAggregateResultState failed job aggregation
+	FailedJobAggregateResultState JobAggregateResultState = "Failed"
```

Comment:
I suspect you'll need running and pending as well.

Reply:
maybe just pending

Reply:
Added Pending.